pkg/volume/rbd/rbd_util.go

```diff
@@ -727,7 +727,12 @@ func getRbdImageSize(output []byte) (int, error) {
 	info := struct {
 		Size int64 `json:"size"`
 	}{}
-	if err := json.Unmarshal(output, &info); err != nil {
+	startJson := strings.LastIndex(string(output), "{")
+	endJson := strings.LastIndex(string(output), "}")
+	if startJson > endJson {
+		return 0, fmt.Errorf("can't find json result")
+	}
+	if err := json.Unmarshal([]byte(string(output)[startJson: endJson+1]), &info); err != nil {
```

Comment:
This way to filter out warning infos is too hacky. And `mount.Exec` does not provide a method to filter out stderr stream too.
How about using `-c /dev/null` to override ceph configuration, then `rbd` command will not search ceph configuration file from the host? So no warnings will be emitted.
Actually, for kubelet rbd plugin, all required information to access ceph cluster are provided from command line arguments.

Reply:
`rbd` command in this plugin will search and use Ceph configuration if it exists on the host is a side-effect, it is not documented and officially supported I guess. I think we should remove it.

cc @rootfs and @jsafrane what's your opinion?

Reply:
I don't understand Ceph enough to judge if it's a common thing to configure its client via config in /etc/ on every node. Are all options really configurable via Kubernetes PV? If we remove parsing of /etc/, how likely is some cluster going to break?

There should be at least some a warning in release notes + deprecation period.

Reply:
It is not a common thing to configure its client via config in `/etc` on every node. It's hard to maintain (need to copy configurations to every new node), and easy to make mistakes. All options are configurable via Kubernetes PV and client arguments will override configurations in `/etc/ceph`.  

In fixing this bug, I don't it is necessary to warn users that we don't support reading configurations from `/etc/ceph` because we have all required arguments in the command line:

https://github.com/kubernetes/kubernetes/blob/ea0099103ea5532b49f5a9e5570d44f33b24bd67/pkg/volume/rbd/rbd_util.go#L702-L704

Reply:
@cofyc @jsafrane Thank you and I make sense. If required arguments in the command line in code at beginning, it will not break exist cluster. And if we make sure all the arguments of the command will override the options of config file, it is safety to do following @cofyc says. But I will also make annotations to attention user not necessary to config  `/etc/ceph` in code.

Reply:
@cofyc I just came across this issue and I did a quick test in my environment, your suggestion to use `-c /dev/null` will still output the following line to stderr:
> 2019-06-14 11:23:20.741527 7f1d737e60c0 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.kube-admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory

If you go that way you should also include `-k /dev/null -K /dev/null` to point the keyrings to the empty file.

Reply:
touch a empty `/etc/ceph/ceph.conf` and `/etc/ceph/keyring`  files should be work for take ceph client warning message away.

Reply:
@Symbianx You are right, but -K will cover the secret, i will change with -k /dev/null to cover it.